File: 3sem/4lab/Banks/models/BankConfigurations/DepositBankAccountConfiguration.cs
using Banks.DepositInterestRateSelectors;
using Banks.Exceptions;

namespace Banks.Models.BankConfigurations;

public record DepositBankAccountConfiguration
{
    public DepositBankAccountConfiguration(decimal unverifiedLimit, TimeSpan duration, IDepositInterestRateSelector selector)
    {
        ArgumentNullException.ThrowIfNull(selector);

        if (unverifiedLimit < 0)
            throw InvalidBankAccountOperationException.OnInvalidCreditLimit();

        UnverifiedLimit = unverifiedLimit;
        Duration = duration;
        Selector = selector;
    }

    public decimal UnverifiedLimit { get; }
    public TimeSpan Duration { get; }
    public IDepositInterestRateSelector Selector { get; }
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/Models/BankConfigurations/DepositBankAccountConfiguration.java
package BanksRemake.Domain.Models.BankConfigurations;

import BanksRemake.Domain.DepositInterestRateSelectors.DepositInterestRateSelector;

import java.time.Period;

/**
 * A set of parameters characterizing the deposit account.
 * @param unverifiedLimit maximum transaction amount for an unverified user.
 * @param duration time to withdraw money.
 * @param selector object that determines the interest rate.
 */
public record DepositBankAccountConfiguration(double unverifiedLimit,
                                              Period duration,
                                              DepositInterestRateSelector selector) {
}
